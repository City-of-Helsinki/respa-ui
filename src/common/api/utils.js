import get from 'lodash/get';

import constants from '../../../app/constants/AppConstants';
import store from '../../store';

let authToken;

store.subscribe(() => {
  const state = store.getState();
  authToken = get(state, 'auth.token');
});

function getAuthorizationHeader() {
  return authToken
    ? { Authorization: `JWT ${authToken}` }
    : {};
}

/**
 * Getter for API url.
 * @param endpoint
 * @param params
 * @param trailingSlash This is because Django adds trailing slash into every endpoint.
 * @returns {string}
 */
export const getUrl = (endpoint, params = {}, trailingSlash = true) => {
  const query = Object.keys(params).map(key => `${key}=${params[key]}`).join('&');
  return `${constants.API_URL}/${endpoint}${trailingSlash ? '/' : ''}${query ? `?${query}` : ''}`;
};

/**
 * Getter for default headers.
 * @returns {object}
 */
export const getHeaders = () => ({
  ...constants.REQUIRED_API_HEADERS,
  ...getAuthorizationHeader(),
});
