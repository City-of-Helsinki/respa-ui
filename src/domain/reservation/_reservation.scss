@import 'manage/page/manageReservationsPage';
@import 'manage/list/manageReservationsList';
@import 'manage/status/manageReservationsStatus';
