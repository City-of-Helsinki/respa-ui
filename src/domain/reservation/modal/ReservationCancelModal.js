import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-bootstrap/lib/Modal';
import { connect } from 'react-redux';
import Toggle from 'react-toggle';

import injectT from '../../../../app/i18n/injectT';
import CompactReservationList from '../../../../app/shared/compact-reservation-list/CompactReservationList';

const mapStateToProps = (state) => {
  return {
    userId: state.auth.userId,
    users: state.data.users
  };
};

const ReservationCancelModal = ({
  onEditReservation, parentToggle, reservation, toggleShow, t, userId, users
}) => {
  const [show, setShow] = useState(toggleShow);
  const [checkboxDisabled, handleCheckbox] = useState(false);
  const handleClose = () => {
    setShow(() => false);
    parentToggle(false);
  };

  /**
   * Both <root>/varaamo/app/shared/modals/reservation-cancel/ReservationCancelModalContainer.js
   * and ReservationCancelModal are using CompactReservationList to output resource name, date, etc.
   * We need to alter reservation a little bit before passing it to CompactReservationList.
   */
  const reservationCopyId = JSON.parse(JSON.stringify(reservation));
  const id = reservationCopyId.resource.id;
  reservationCopyId.resource = id;

  const reservationCopy = JSON.parse(JSON.stringify(reservation));
  const name = reservationCopy.resource.name.fi;
  reservationCopy.resource.name = name;

  useEffect(() => {
    setShow(toggleShow);
  }, [toggleShow]);

  const renderCheckBox = (notice, onConfirm) => {
    return (
      <div>
        <p><strong>{notice}</strong></p>
        <Toggle
          defaultChecked={false}
          id="checkbox"
          onChange={e => onConfirm(e.target.checked)}
        />
      </div>
    );
  };

  return (
    <Modal
      onHide={handleClose}
      show={show}
    >
      <Modal.Header closeButton>
        <Modal.Title>{t('ReservationCancelModal.cancelAllowedTitle')}</Modal.Title>
      </Modal.Header>

      <Modal.Body>
        <div>
          <p><strong>{t('ReservationCancelModal.lead')}</strong></p>
          {reservation.resource
            && (
              <CompactReservationList
                reservations={[reservationCopyId]}
                resources={{ [reservationCopy.resource.id]: reservationCopy.resource }}
              />
            )
          }
          {
            reservation.resource
            && renderCheckBox(
              t('ReservationInformationForm.refundCheckBox'),
              () => console.log('--- FOO ---')
            )
          }
        </div>
      </Modal.Body>

      <Modal.Footer>
        Modal footer
      </Modal.Footer>
    </Modal>
  );
};

ReservationCancelModal.propTypes = {
  onEditReservation: PropTypes.func.isRequired,
  parentToggle: PropTypes.func.isRequired,
  reservation: PropTypes.object.isRequired,
  toggleShow: PropTypes.bool.isRequired,
  t: PropTypes.func.isRequired,
  userId: PropTypes.string.isRequired,
  users: PropTypes.object.isRequired
};

export default connect(mapStateToProps)(injectT(ReservationCancelModal));
