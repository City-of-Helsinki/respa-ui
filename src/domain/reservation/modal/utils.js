export const getCancelCategories = (data = [], isAdmin, locale) => {
  return data
    .filter(({ reservation_type: reservationType }) => (
      isAdmin ? true : reservationType === 'own'))
    .map(category => ({
      value: category.id,
      label: category.name[locale || 'fi'],
    }));
};
