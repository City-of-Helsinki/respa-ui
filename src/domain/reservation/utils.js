import moment from 'moment';
import find from 'lodash/find';
import sortBy from 'lodash/sortBy';
import clone from 'lodash/clone';
import tail from 'lodash/tail';
import last from 'lodash/last';

export const combineReservations = (reservations) => {
  if (!reservations || !reservations.length) {
    return [];
  }

  const sorted = sortBy(reservations, 'begin');
  const initialValue = [clone(sorted[0])];

  return tail(sorted).reduce((previous, current) => {
    if (current.begin === last(previous).end) {
      last(previous).end = current.end;
    } else {
      previous.push(clone(current));
    }
    return previous;
  }, initialValue);
};

export const getNextAvailableTime = (reservations, fromMoment = moment()) => {
  const combinedReservations = combineReservations(reservations);
  if (!combinedReservations.length || fromMoment < moment(combinedReservations[0].begin)) {
    return fromMoment;
  }

  const ongoingReservation = find(combinedReservations, reservation => (
    moment(reservation.begin) <= fromMoment && fromMoment < moment(reservation.end)
  ));

  return ongoingReservation ? moment(ongoingReservation.end) : fromMoment;
};

export const getCurrentReservation = (reservations) => {
  const now = moment();
  return find(reservations, (reservation) => {
    return moment(reservation.begin) < now && now < moment(reservation.end);
  });
};
