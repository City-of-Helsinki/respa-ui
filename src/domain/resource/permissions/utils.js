import difference from 'lodash/difference';

import { resourceRoles, resourcePermissionTypes, resourcePermissionsByRole } from './constants';

// Returns the role with the most permissions or null.
export function roleMapper(isAdmin, isManager, isViewer) {
  if (isAdmin) {
    return resourceRoles.UNIT_ADMINISTRATOR;
  }

  if (isManager) {
    return resourceRoles.UNIT_MANAGER;
  }

  if (isViewer) {
    return resourceRoles.UNIT_VIEWER;
  }

  return null;
}

const permissionOptions = Object.values(resourcePermissionTypes);
const allValuesInArray = (subset, superset) => difference(subset, superset).length === 0;

export function hasPermissionForResource(resourceRole, requiredPermissions) {
  // undefined roles don't receive any permissions implicitly
  if (resourceRole === undefined || resourceRole === null) {
    return false;
  }

  if (!Object.values(resourceRoles).includes(resourceRole)) {
    throw Error(`resourceRole needs to be one of: ${Object.values(resourceRoles).join(', ')}.`);
  }

  if (!Array.isArray(requiredPermissions) && typeof requiredPermissions !== 'string') {
    throw Error('requiredPermissions has to be either a string or an array.');
  }

  if (
    Array.isArray(requiredPermissions)
    && !allValuesInArray(requiredPermissions, permissionOptions)
  ) {
    const unknownPermissions = difference(requiredPermissions, permissionOptions);

    // eslint-disable-next-line no-console
    console.warn([
      `Permission asked with unknown permission: ${unknownPermissions.join(', ')}.`,
      `Permission needs to be on of: ${permissionOptions.join(', ')}.`,
    ].join('\n'));
  }

  if (typeof requiredPermissions === 'string' && !permissionOptions.includes(requiredPermissions)) {
    // eslint-disable-next-line no-console
    console.warn([
      `Permission asked with unknown permission: ${requiredPermissions}.`,
      `Permission needs to be on of: ${permissionOptions.join(', ')}.`,
    ].join('\n'));
  }

  const rolePermissions = resourcePermissionsByRole[resourceRole];

  if (Array.isArray(requiredPermissions)) {
    return allValuesInArray(requiredPermissions, rolePermissions);
  }

  return rolePermissions.includes(requiredPermissions);
}
