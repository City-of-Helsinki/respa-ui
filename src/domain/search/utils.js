import get from 'lodash/get';
import range from 'lodash/range';
import capitalize from 'lodash/capitalize';
import sortBy from 'lodash/sortBy';

import constants from '../../../app/constants/AppConstants';

// TODO: Change to use query-string module.
export const getFiltersFromUrl = (location) => {
  const query = new URLSearchParams(location.search);
  const filters = {};

  query.forEach((value, key) => {
    if (constants.SUPPORTED_SEARCH_FILTERS[key] !== undefined) {
      filters[key] = value;
    }
  });

  return filters;
};

// TODO: Change to use query-string module.
export const getSearchFromFilters = (filters) => {
  const query = new URLSearchParams(filters);
  return query.toString();
};

export const getUnitOptions = (units, locale) => {
  const options = units.map(unit => ({
    value: unit.id,
    label: get(unit, `name[${locale}]`, ''),
  }));

  return sortBy(options, 'label');
};

export const getPurposeOptions = (purposes, locale) => {
  const options = purposes
    .filter(purpose => purpose.parent === null)
    .map(unit => ({
      value: unit.id,
      label: get(unit, `name[${locale}]`, ''),
    }));

  return sortBy(options, 'label');
};

/**
 * Getter for people capacity options.
 * @returns {{label: number, value: number}[]}
 */
export const getPeopleCapacityOptions = () => {
  return [
    ...range(1, 10),
    ...range(10, 35, 5),
    ...range(40, 110, 10),
  ].map(number => ({ label: number, value: number }));
};

/**
 * Getter for municipality options.
 * @returns {{label: string, value: string}[]}
 */
export const getMunicipalityOptions = () => {
  let municipalities = constants.DEFAULT_MUNICIPALITY_OPTIONS;

  if (Array.isArray(SETTINGS.CUSTOM_MUNICIPALITY_OPTIONS)
    && SETTINGS.CUSTOM_MUNICIPALITY_OPTIONS.length) {
    municipalities = SETTINGS.CUSTOM_MUNICIPALITY_OPTIONS;
  }

  return municipalities.map((municipality) => {
    const municipalityStr = typeof municipality === 'string' ? municipality : municipality.toString();

    return {
      value: municipalityStr.toLowerCase(),
      label: capitalize(municipalityStr),
    };
  });
};
