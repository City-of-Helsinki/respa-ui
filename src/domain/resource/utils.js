import round from 'lodash/round';
import filter from 'lodash/filter';
import findIndex from 'lodash/findIndex';
import get from 'lodash/get';
import find from 'lodash/find';
import moment from 'moment';

import * as urlUtils from '../../common/url/utils';
import * as dataUtils from '../../common/data/utils';
import * as reservationUtils from '../reservation/utils';
import constants from '../../../app/constants/AppConstants';
import forEach from "lodash/forEach";


/**
 * Getter for resource page link.
 * Returns a resource page url with the given query params.
 * @param resource
 * @param query
 * @returns {string}
 */
export const getResourcePageLink = (resource, query) => {
  return urlUtils.getLinkString(`/resources/${resource.id}`, query);
};

/**
 * Getter for formatted unit address string.
 * @param unit
 * @param locale
 * @returns {string}
 */
export const getUnitAddress = (unit, locale) => {
  let postal;
  const streetAddress = dataUtils.getLocalizedFieldValue(unit.street_address, locale);
  if (unit && unit.address_zip) {
    postal = unit.address_zip;
  }
  if (unit && unit.municipality) {
    postal = postal ? `${postal} ${unit.municipality}` : unit.municipality;
  }

  return `${streetAddress}${postal ? ` ${postal}` : ''}`;
};

/**
 * Getter for formatted resource distance string.
 * @param resource
 * @returns {string}
 */
export const getResourceDistance = (resource) => {
  const km = resource.distance / 1000;
  return km ? `${round(km, km < 10 ? 1 : null)} km` : '';
};

/**
 * Getter for price string used in resource cards.
 * @param resource
 * @param t
 * @returns {string|*}
 */
export const getPrice = (resource, t) => {
  const minPricePerHour = resource.min_price_per_hour;
  const maxPricePerHour = resource.max_price_per_hour;

  if (!(minPricePerHour || maxPricePerHour)) {
    return t('ResourceIcons.free');
  }

  if (minPricePerHour && maxPricePerHour && minPricePerHour !== maxPricePerHour) {
    return `${Number(minPricePerHour)} - ${Number(maxPricePerHour)} €/h`;
  }

  const priceString = maxPricePerHour || minPricePerHour;
  const price = priceString !== 0 ? Number(priceString) : 0;
  if (price === 0) {
    return t('ResourceIcons.free');
  }

  return price ? `${price} €/h` : null;
};

export const isFree = (resource) => {
  const minPricePerHour = resource.min_price_per_hour;
  const maxPricePerHour = resource.max_price_per_hour;

  if (!(minPricePerHour || maxPricePerHour)) {
    return true;
  }

  if (minPricePerHour && maxPricePerHour && minPricePerHour !== maxPricePerHour) {
    return false;
  }

  const priceString = maxPricePerHour || minPricePerHour;
  const price = priceString !== 0 ? Number(priceString) : 0;

  return price === 0;
};

export const getOpeningHours = (resource, date = null) => {
  if (!resource) {
    return null;
  }

  const openingHours = get(resource, 'opening_hours', []);
  if (!openingHours.length) {
    return null;
  }

  const index = date ? findIndex(openingHours, item => item.date === date) : 0;
  return {
    closes: get(openingHours, `[${index}].closes`),
    opens: get(openingHours, `[${index}].opens`),
  };
};

const reservingIsRestricted = (resource, date) => {
  if (!date) {
    return false;
  }

  const isAdmin = get(resource, 'user_permissions.is_admin', false);
  const isLimited = resource.reservable_before && moment(resource.reservable_before).isBefore(moment(date), 'day');

  return isLimited && !isAdmin;
};

export const getOpenReservations = (resource) => {
  return filter(resource.reservations, (reservation) => {
    return reservation.state !== 'cancelled' && reservation.state !== 'denied';
  });
};

export const getAvailabilityDataForNow = (resource, date) => {
  const openingHours = getOpeningHours(resource, date);
  const reservations = getOpenReservations(resource);

  if (!openingHours || !openingHours.closes || !openingHours.opens) {
    return { status: 'closed', bsStyle: 'danger' };
  }

  const nowMoment = moment();
  const opensMoment = moment(openingHours.opens);
  const closesMoment = moment(openingHours.closes);
  const beginMoment = nowMoment > opensMoment ? nowMoment : opensMoment;

  if (nowMoment > closesMoment) {
    return { status: 'closed', bsStyle: 'danger' };
  }

  const currentReservation = reservationUtils.getCurrentReservation(reservations);
  if (currentReservation || nowMoment < opensMoment) {
    const nextAvailableTime = reservationUtils.getNextAvailableTime(reservations, beginMoment);
    if (nextAvailableTime < closesMoment) {
      return {
        status: 'availableAt',
        bsStyle: 'danger',
        values: { time: nextAvailableTime.format(constants.TIME_FORMAT) },
      };
    }
    return { status: 'reserved', bsStyle: 'danger' };
  }

  return { status: 'available', bsStyle: 'success' };
};

export const getAvailabilityDataForWholeDay = (resource, date) => {
  const openingHours = getOpeningHours(resource, date);

  if (!openingHours || !openingHours.closes || !openingHours.opens) {
    return { status: 'closed', bsStyle: 'danger' };
  }

  if (reservingIsRestricted(resource, date)) {
    return { status: 'reservingRestricted', bsStyle: 'danger' };
  }

  const reservations = getOpenReservations(resource);

  const opensMoment = moment(openingHours.opens);
  const closesMoment = moment(openingHours.closes);
  let total = closesMoment - opensMoment;

  forEach(reservations, (reservation) => {
    const resBeginMoment = moment(reservation.begin);
    if (!resBeginMoment.isSame(opensMoment, 'd')) {
      return;
    }

    const resEndMoment = moment(reservation.end);
    total -= resEndMoment;
    total += resBeginMoment;
  });

  const asHours = moment.duration(total).asHours();
  const rounded = Math.ceil(asHours * 2) / 2;

  if (rounded === 0) {
    return { status: 'reserved', bsStyle: 'danger' };
  }

  return {
    status: 'availableTime',
    bsStyle: 'success',
    values: { hours: rounded },
  };
};
