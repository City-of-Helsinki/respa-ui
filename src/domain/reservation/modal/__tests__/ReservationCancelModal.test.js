import React from 'react';
import toJSON from 'enzyme-to-json';
import Modal from 'react-bootstrap/lib/Modal';
import Button from 'react-bootstrap/lib/Button';

import { shallowWithIntl, globalDateMock } from '../../../../../app/utils/testUtils';
import ReservationCancelModal from '../ReservationCancelModal';
import reservation from '../../../../common/data/fixtures/reservation';
import CompactReservationList from '../../../../../app/shared/compact-reservation-list/CompactReservationList';

describe('domain/reservation/modal/ReservationCancelModal', () => {
  const mockReservation = reservation.build({ begin: '2019-08-14T14:00:00+03:00', end: '2019-08-14T15:00:00+03:00' });
  const props = {
    t: jest.fn(),
    billable: false,
    onEditReservation: jest.fn(),
    parentToggle: jest.fn(),
    reservation: mockReservation,
    toggleShow: true,
    cancelCategories: [],
  };

  const getWrapper = (extraProps = {}) => {
    return shallowWithIntl(<ReservationCancelModal {...props} {...extraProps} />);
  };

  beforeEach(() => {
    jest.restoreAllMocks();
  });

  describe('Modal', () => {
    test('renders correctly', () => {
      const extraProps = {
        userId: 'foo',
        users: { bar: 'bar' },
      };
      globalDateMock();
      const wrapper = shallowWithIntl(
        <ReservationCancelModal {...props} {...extraProps} />,
      );
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders a checkbox when the resource is billable', () => {
      globalDateMock();
      const extraProps = {
        userId: 'foo',
        users: { bar: 'bar' },
        billable: true,
      };
      const wrapper = shallowWithIntl(
        <ReservationCancelModal {...props} {...extraProps} />,
      );
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Modal header', () => {
    function getModalHeaderWrapper(extraProps) {
      return getWrapper(extraProps).find(Modal.Header);
    }

    test('is rendered', () => {
      expect(getModalHeaderWrapper()).toHaveLength(1);
    });

    test('contains a close button', () => {
      expect(getModalHeaderWrapper().props().closeButton).toBe(true);
    });

    describe('title', () => {
      test('is correct if cancel is allowed', () => {
        const modalTitle = getModalHeaderWrapper({ cancelAllowed: true }).find(Modal.Title);
        expect(modalTitle.length).toBe(1);
        expect(modalTitle.prop('children')).toBe('ReservationCancelModal.cancelAllowedTitle');
      });
    });
  });

  describe('Modal body', () => {
    function getModalBodyWrapper(extraProps) {
      return getWrapper(extraProps).find(Modal.Body);
    }

    test('is rendered', () => {
      expect(getModalBodyWrapper).toHaveLength(1);
    });

    describe('if cancel is allowed', () => {
      const cancelAllowed = true;

      test('renders CompactReservationList', () => {
        expect(
          getModalBodyWrapper({ cancelAllowed }).find(CompactReservationList),
        ).toHaveLength(1);
      });

      test('does not render responsibleContactInfo', () => {
        expect(
          getModalBodyWrapper({ cancelAllowed }).find('.responsible-contact-info'),
        ).toHaveLength(0);
      });
    });

    describe('if resource has payment support', () => {
      test('renders custom message', () => {
        const wrapper = getModalBodyWrapper({
          resource: {
            products: [
              {
                type: 'rent',
              },
            ],
          },
        });

        expect(toJSON(wrapper)).toMatchSnapshot();
      });
    });
  });

  describe('Footer buttons', () => {
    function getFooterButtonsWrapper(extraProps) {
      return getWrapper(extraProps).find(Modal.Footer).find(Button);
    }

    describe('if cancel is allowed', () => {
      const cancelAllowed = true;
      const buttons = getFooterButtonsWrapper({ cancelAllowed });

      test('renders cancel button', () => {
        expect(buttons.at(0).props().children).toBe('ReservationCancelModal.cancelAllowedCancel');
      });

      test('renders confirm button', () => {
        expect(buttons.at(1).props().children).toBe('ReservationCancelModal.cancelAllowedConfirm');
      });
    });
  });
});
