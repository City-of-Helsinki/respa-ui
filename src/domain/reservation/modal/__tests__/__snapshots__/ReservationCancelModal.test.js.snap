// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`domain/reservation/modal/ReservationCancelModal Modal body if resource has payment support renders custom message 1`] = `
<ModalBody
  bsClass="modal-body"
  componentClass="div"
>
  <div>
    <p>
      <strong>
        ReservationCancelModal.lead
      </strong>
    </p>
    <CompactReservationList
      reservations={
        Array [
          Object {
            "begin": "2019-08-14T14:00:00+03:00",
            "comments": "",
            "end": "2019-08-14T15:00:00+03:00",
            "event_description": "",
            "event_subject": "",
            "has_catering_order": false,
            "id": 1,
            "its_own": false,
            "resource": undefined,
            "staff_event": false,
            "state": "requested",
            "url": "https://respa.koe.hel.ninja/v1/reservation/192388/",
            "user": Object {},
            "user_permissions": null,
          },
        ]
      }
      resources={
        Object {
          "undefined": Object {
            "name": "",
          },
        }
      }
    />
  </div>
  <InjectT(SelectField)
    id="cancelReasonCategory"
    label="ReservationInformationForm.cancellationReason"
    onChange={[Function]}
    options={Array []}
  />
  <TextAreaField
    id="cancelReasonExplanation"
    label="ReservationInformationForm.explanation"
    onChange={[Function]}
    rows={5}
    value=""
  />
</ModalBody>
`;

exports[`domain/reservation/modal/ReservationCancelModal Modal renders a checkbox when the resource is billable 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      ReservationCancelModal.cancelAllowedTitle
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div>
      <p>
        <strong>
          ReservationCancelModal.lead
        </strong>
      </p>
      <CompactReservationList
        reservations={
          Array [
            Object {
              "begin": "2019-08-14T14:00:00+03:00",
              "comments": "",
              "end": "2019-08-14T15:00:00+03:00",
              "event_description": "",
              "event_subject": "",
              "has_catering_order": false,
              "id": 1,
              "its_own": false,
              "resource": undefined,
              "staff_event": false,
              "state": "requested",
              "url": "https://respa.koe.hel.ninja/v1/reservation/192388/",
              "user": Object {},
              "user_permissions": null,
            },
          ]
        }
        resources={
          Object {
            "undefined": Object {
              "name": "",
            },
          }
        }
      />
      <div>
        <p>
          <strong>
            ReservationInformationForm.refundCheckBox
          </strong>
        </p>
        <Toggle
          defaultChecked={false}
          icons={
            Object {
              "checked": <Unknown />,
              "unchecked": <Unknown />,
            }
          }
          id="checkbox"
          onChange={[Function]}
        />
      </div>
    </div>
    <InjectT(SelectField)
      id="cancelReasonCategory"
      label="ReservationInformationForm.cancellationReason"
      onChange={[Function]}
      options={Array []}
    />
    <TextAreaField
      id="cancelReasonExplanation"
      label="ReservationInformationForm.explanation"
      onChange={[Function]}
      rows={5}
      value=""
    />
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      ReservationCancelModal.cancelAllowedCancel
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="danger"
      disabled={true}
      onClick={[Function]}
    >
      ReservationCancelModal.cancelAllowedConfirm
    </Button>
  </ModalFooter>
</Modal>
`;

exports[`domain/reservation/modal/ReservationCancelModal Modal renders correctly 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      ReservationCancelModal.cancelAllowedTitle
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div>
      <p>
        <strong>
          ReservationCancelModal.lead
        </strong>
      </p>
      <CompactReservationList
        reservations={
          Array [
            Object {
              "begin": "2019-08-14T14:00:00+03:00",
              "comments": "",
              "end": "2019-08-14T15:00:00+03:00",
              "event_description": "",
              "event_subject": "",
              "has_catering_order": false,
              "id": 1,
              "its_own": false,
              "resource": undefined,
              "staff_event": false,
              "state": "requested",
              "url": "https://respa.koe.hel.ninja/v1/reservation/192388/",
              "user": Object {},
              "user_permissions": null,
            },
          ]
        }
        resources={
          Object {
            "undefined": Object {
              "name": "",
            },
          }
        }
      />
    </div>
    <InjectT(SelectField)
      id="cancelReasonCategory"
      label="ReservationInformationForm.cancellationReason"
      onChange={[Function]}
      options={Array []}
    />
    <TextAreaField
      id="cancelReasonExplanation"
      label="ReservationInformationForm.explanation"
      onChange={[Function]}
      rows={5}
      value=""
    />
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      ReservationCancelModal.cancelAllowedCancel
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="danger"
      disabled={true}
      onClick={[Function]}
    >
      ReservationCancelModal.cancelAllowedConfirm
    </Button>
  </ModalFooter>
</Modal>
`;
